File: 백준/Silver/1268. 임시 반장 정하기/임시 반장 정하기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int n;
    static int[][] classInfo;
    static boolean[][] meetInfo;
    static BufferedReader br;
    static StringTokenizer st;


    public static void main(String[] args) throws Exception {
        br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());

        classInfo = new int[n][5];
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < 5; j++) {
                classInfo[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        meetInfo = new boolean[n][n];
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = j; k < n; k++) {
                    if (classInfo[j][i] == classInfo[k][i]) {
                        meetInfo[j][k] = true;
                        meetInfo[k][j] = true;
                    }
                }
            }
        }

        int maxCnt = 0;
        int maxStudent = 0;
        for (int currentStudent = 0; currentStudent < n; currentStudent++) {
            int currentCnt = 0;
            for (int i = 0; i < n; i++) {
                if (meetInfo[currentStudent][i]) {
                    currentCnt++;
                }
            }

            if (currentCnt > maxCnt) {
                maxCnt = currentCnt;
                maxStudent = currentStudent;
            }
        }
        System.out.println(maxStudent + 1);
    }
}

File: 백준/Silver/1268. 임시 반장 정하기/임시 반장 정하기.py
from sys import stdin

input = stdin.readline

n = int(input())
class_info = [list(map(int, input().rsplit())) for _ in range(n)]

meet_info = [[False for _ in range(n)] for _ in range(n)]

for i in range(5):
  for j in range(n):
    for k in range(j, n):
      if class_info[j][i] == class_info[k][i]:
        meet_info[j][k] = 1
        meet_info[k][j] = 1

max_cnt = 0
max_student = 0
for cur_student in range(n):
  cur_cnt = sum(meet_info[cur_student])
  if max_cnt >= cur_cnt:
    continue

  else:
    max_cnt = cur_cnt
    max_student = cur_student

print(max_student+1)
  
    